zilliqa/src/scilla.rs

```diff
@@ -775,7 +787,19 @@ impl ActiveCall {
         Ok(())
     }
 
-    fn fetch_blockchain_info(&self, name: String, args: String) {
-        eprintln!("fetch_blockchain_info - {name}, {args}");
+    fn fetch_blockchain_info(&self, name: String, _args: String) -> Result<String> {
+        match name.as_str() {
+            "CHAINID" => Ok(self.state.network_id().to_string()),
+            "BLOCKNUMBER" => Ok(self.current_block.number.to_string()),
+            "TIMESTAMP" => Ok(self
+                .current_block
+                .timestamp
+                .duration_since(SystemTime::UNIX_EPOCH)?
+                .as_micros()
+                .to_string()),
```

Comment:
Double checking that Scilla expects microseconds here?

Reply:
Actually, the `BLOCKNUMBER`, `BLOCKHASH` and `TIMESTAMP` refers to the value of second argument passed in the function and not the current block. so we would need to interpret `_args` and read the persistence to read those parameters.Here is the zq1 implementation - https://github.com/Zilliqa/Zilliqa/blob/ba6dc3d1cd08de09ea9f543340c983d83ecdfde7/src/libScilla/ScillaIPCServer.cpp#L516

Reply:
@chetan-zilliqa Thanks! Good catch! :) But the `BLOCKNUMBER` seems to have zero-param

Reply:
Yeah, you are correct. `BLOCKNUMBER` is the current block number..